aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncPaginatedResult.java

```diff
@@ -27,7 +28,7 @@
     private final Iterable<T> items;
 
     AppSyncPaginatedResult(@NonNull Iterable<T> items,
-                           @NonNull GraphQLRequest<PaginatedResult<T>> requestForNextResult) {
+                           @Nullable GraphQLRequest<PaginatedResult<T>> requestForNextResult) {
```

Comment:
Is it possible to have a paginated result, that has no request associated with it?

Reply:
Yes, if `nextToken` is null (the last page), then the `requestForNextResult` will also be null